zerver/lib/actions.py

```diff
@@ -7281,11 +7281,12 @@ def do_resend_user_invite_email(prereg_user: PreregistrationUser) -> int:
 
     check_invite_limit(prereg_user.referred_by.realm, 1)
 
-    prereg_user.invited_at = timezone_now()
-    prereg_user.save()
     invite_expires_in_days = (
         prereg_user.confirmation.get().expiry_date - prereg_user.invited_at
     ).days
+
+    prereg_user.invited_at = timezone_now()
+    prereg_user.save()
```

Comment:
Can you adjust the backend tests for this codepath to verify the new invitation time and expiration date for the resulting final invitation?  

It is probably also worth adding a bit of a block comment explaining why this function results in the correct behavior.  E.g. I think it's the case that we're resetting the confirmation object as though it was sent today, and then retriggering the invitation/confirmation email, which should get an expiration time the default number of days in the future -- is that right?

Reply:
Yes you are right it needs an explanation so that the bug does not return in future changes.

Regarding the backend test I'm not sure which file I should look at? All I could find was a simple test (that looked fine to me) in ```test_counts.py```

Reply:
I found `test_successful_resend_invitation` in `test_signup.py` by with `git grep resend`; I think it's likely that's what you want to extend.

(Probably no need for a new test, just additional assertions in the existing one)

Reply:
Awesome, I added the necessary assertion in ```test_successful_resend_invitation``` and it confirms that it does indeed work (that is if I manage to write the test correctly).